ibis/backends/base/sqlglot/dialects.py

```diff
@@ -68,6 +69,45 @@ class Generator(Postgres.Generator):
         }
 
 
+def _calculate_precision(interval_value: int) -> int:
+    """Calculate interval precision.
+
+    FlinkSQL interval data types use leading precision and fractional-
```

Comment:
Is this file for the base sqlglot dialect? It looks like we are adding flink-specific logic into the base implementation? It may be good to get @cpcloud or @jcrist 's feedback on a better design pattern to use here.

Reply:
Cool, thanks for the note. Sorry got busy and couldn't review as often as I would like, I will have to rely more on the team :)

Reply:
No worries! ;)